core/src/main/java/org/apache/iceberg/rest/RESTUtil.java

```diff
@@ -38,6 +40,12 @@ public class RESTUtil {
   private static final Joiner NAMESPACE_ESCAPED_JOINER = Joiner.on(NAMESPACE_ESCAPED_SEPARATOR);
   private static final Splitter NAMESPACE_ESCAPED_SPLITTER =
       Splitter.on(NAMESPACE_ESCAPED_SEPARATOR);
+  public static final Function<Map<String, String>, RESTClient> DEFAULT_CLIENT_BUILDER =
```

Comment:
this shouldn't be necessary. Can you please revert the refactorings from 1536b7f7e6276a15527a5fec6bb16ec0f47544b5 and only add the new tests?